### ISSUE TITLE
Proposal: deprecate modrescdatapath by only setting rescpath once, on the resource

### ISSUE THREAD

**Issue by bh9** [irods/irods#4738]
## Feature
Currently, when a resource changes path, you have to update the path of every file on the resource. This adds a layer of complexity, and discourages moving of resources. If the data objects didn't include the path of the resource, then resources could be moved around freely, only ever having to change a single entry. 
The most obvious downside is that anything that needed the path would also have to look up the resource path and fit them together.
Currently:
```
DATA_PATH=/resc/path/data/path
RESC_VAULT_PATH=/resc/path
```
Proposed:
```
DATA_PATH=/data/path
RESC_VAULT_PATH=/resc/path
```

**Comment by trel** [comment]
Agreed - this would make administration easier and be cleaner in the database.

However, this is currently optimized for read, rather than write.  Changing this would require additional joins for nearly all operations.

A full solution to this would involve changing the table structure of iRODS more broadly to fully separate the logical (data objects and collections) from the physical (replicas representing files on disk).   A change like that would have deep implications for GenQuery and its interfaces as well.
